core/state/statedb.go

```diff
@@ -368,6 +368,30 @@ func (s *StateDB) HasSelfDestructed(addr common.Address) bool {
 	return false
 }
 
+// CheckTransactionConditional validates the account preconditions against the statedb.
+func (s *StateDB) CheckTransactionConditional(cond *types.TransactionConditional) error {
+	for addr, acct := range cond.KnownAccounts {
+		if root, isRoot := acct.Root(); isRoot {
+			storageRoot := s.GetStorageRoot(addr)
```

Comment:
Note that the state may change while this loop is running, since no lock is being held over the state-db, and it's not pinned to some snapshot either.

Reply:
Added a comment to the docstring that the caller is responsible that no changes are made while this function is called. All uses are snapshots so we're good